### ISSUE TITLE
Reconnection problem

### ISSUE THREAD

**Issue by jesusvpct** [tmate-io/tmate#168]
Hi! I have 2.2.1 version installed on a Raspberry Pi and I have a problem with reconnection. When I lost connection from internet for some minutes, reconnection works perfect, but if I lost connection more time, like 30 minutes or more, when I try to connect I have a message like this:
```
The session host disconnected 52 minutes ago.
Hopefully it will reconnect soon. You may try again later.
Connection to lon1.tmate.io closed.
```

If I enter to the Raspberry and I see connections with netstat, there is already one, and if I test with curl -L tmate.io/api/t/{my_session_id}, it shows that is not closed.

To get reconnection work, I have to execute this command: `tmate -S /tmp/tmate-sock attach-session`

Is there any solution to get it work without executing that command on the device? This device will be accessible only by remote connection, so It's critical.

Thanks!

**Comment by nviennot** [comment]
Hello Jesús,

I have been pushing a few updates on tmate today. I fixed a long standing
reconnection bug.
install the newest version of tmate by compiling the github sources. If you
don't know how to do that, let me know and I'll help.
I hope it will fix the problem. Also, you can try the new foreground mode,
you can now launch tmate with "tmate -F". I made this to handle the remote
server usecase.

If the problem persists, please let me know, I want to eradicate these
sorts of issues.

Thanks,
Nico

**Comment by jesusvpct** [comment]
Thanks! I'll try it in next days. I'll write you any feedback!

Thanks again.

**Comment by jesusvpct** [comment]
I've compiled tmate on the raspberry, but it has the same problem. Cannot connect when internet is getting back. I'm running: `tmate -F -S /tmp/tmate.sock new-session -d`

Perhaps I've doing something wrong?

Thanks.

**Comment by nviennot** [comment]
The command is okay
I don't know how this bug is happening, I can't reproduce the issue.

Does the message "Reconnecting..." still happens when the internet is back?
Does tmate freezes?
Can you provide a stack dump? (`gdb -p $(pgrep tmate)` then `bt`)
Also, can you send me a core dump with `gcore $(pgrep tmate)` when the bug happens?
You can send it to tmate@viennot.com

This is a critical bug, I'd like to fix it.

**Comment by jesusvpct** [comment]
Thanks for your response. I'll do your suggestions. Until then, the problem was the same. I had to execute the command `tmate -S /tmp/tmate-sock attach-session` to get it work again. Internet was offline for 1 hour I think, then it comes back but when tried to connect I get the message:
```
The session host disconnected 46 minutes ago.
Hopefully it will reconnect soon. You may try again later.
Connection to lon1.tmate.io closed.
```

But I'll give you any new feedback as soon I can with your steps.

Thanks again!

**Comment by nviennot** [comment]
Fixed by c71307ed5c9c00c1fdd28add86b5d06a7fceb98e
